src/string.rs

```diff
@@ -223,6 +224,24 @@ fn not_whitespace_except_line_feed(g: &str) -> bool {
 /// FIXME(issue#3281): We must follow UAX#14 algorithm instead of this.
 fn break_string(max_width: usize, trim_end: bool, line_end: &str, input: &[&str]) -> SnippetState {
     let break_at = |index /* grapheme at index is included */| {
+        // Ensure break is not after an escape '\' as it will "escape" the `\` that is added
+        // for concatenating the two parts of the broken line.
+        let index = if input[index] != "\\" {
+            index
+        } else {
+            let index_offset = match input[0..index]
+                .iter()
+                .rposition(|grapheme| grapheme.ne(&"\\"))
+            {
+                // There is a non-`\` to the left
+                Some(non_backslash_index) => (index - non_backslash_index) % 2,
+                // Only `\` to the left
+                None => (index + 1) % 2,
+            };
+            // Make sure break is after even number (including zero) of `\`
```

Comment:
I wonder if you could elaborate more on these comments or rephrase them somehow. When I read them I feel like there are some details that are missing. For example, why do we need to make sure the break is after an even number?

Reply:
Enhanced the comments and moved some to a more appropriate place.  I hope they are sufficiently clear now.